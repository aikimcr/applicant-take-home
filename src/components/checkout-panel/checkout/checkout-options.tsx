import React from 'react';
import { useDispatch } from 'react-redux';
import { useAppSelector } from '../../../hooks';
import { getCurrentGiftCardIndex, PrizeoutOffer, setGiftCardIndex } from '../../../slices/offers-slice';
import { AppDispatch } from '../../../store';
import CheckoutCostSummary from './checkout-cost-summary';
import CheckoutOptionsButtons from './checkout-options-buttons';

import './checkout.less';

interface CheckoutOptionsProps {
    offer: PrizeoutOffer;
}

export const CheckoutOptions: React.FC<CheckoutOptionsProps> = ({ offer }): React.ReactElement => {
    const giftCardIndex = useAppSelector(getCurrentGiftCardIndex);
    const dispatch = useDispatch<AppDispatch>();

    const newGiftCardIndex = (newIndex: number) => {
        dispatch(setGiftCardIndex(newIndex));
    };

    return (
        <section className="checkout-options">
            <div className="grid grid--top-bottom grid-stretch-top">
                <div className="grid__item">
                    <header>Select Redemption Amount</header>
                </div>
                <div className="grid__item">
                    <section className="options-list">
                        <CheckoutOptionsButtons giftcardList={offer.giftcard_list} onNewGiftCard={newGiftCardIndex} />
                    </section>
                </div>
                <div className="grid__item">
                    <CheckoutCostSummary giftcard={offer.giftcard_list[giftCardIndex]} />
                </div>
            </div>
        </section>
    );
};

export default CheckoutOptions;
