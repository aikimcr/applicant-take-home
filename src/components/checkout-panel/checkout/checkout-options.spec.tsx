import { render, screen } from '../../../testing/test-utils';
import React from 'react';
import '@testing-library/jest-dom';

import { CheckoutOptions } from './checkout-options';
import { PrizeoutOffer } from '../../../slices/offers-slice';

describe('Test CheckoutOptions', () => {
    const testOffer: PrizeoutOffer = {
        checkout_hero_url: null,
        currency_code: 'USD',
        description: '0',
        giftcard_list: [
            {
                checkout_value_id: '775af84c-d388-4da7-b4cf-89f7fee1ebda',
                cost_in_cents: 1470,
                display_bonus: 2.04,
                display_monetary_bonus: null,
                value_in_cents: 1500,
            },
            {
                checkout_value_id: '21a2b9c5-f668-4c04-907b-019c53d72074',
                cost_in_cents: 2450,
                display_bonus: 2.04,
                display_monetary_bonus: null,
                value_in_cents: 2500,
            },
            {
                checkout_value_id: '24ed1a3d-8d86-4c7f-8bc8-1f4c3e6399b8',
                cost_in_cents: 4900,
                display_bonus: 2.04,
                display_monetary_bonus: null,
                value_in_cents: 5000,
            },
        ],
        image_url: 'https://d13080yemosbe2.cloudfront.net/Images/GiftCardFaceplates/External/MAGZ-D-00-16-1_fp01.png',
        is_enabled: true,
        logomark_url: null,
        name: 'Magazines.com',
        stores: [],

        support_creative_list: [],
        tag: null,
    };

    test('Component matches snapshot', () => {
        const newCardHandler = jest.fn();
        const { asFragment } = render(<CheckoutOptions offer={testOffer} />);
        expect(asFragment()).toMatchSnapshot();
    });
});
