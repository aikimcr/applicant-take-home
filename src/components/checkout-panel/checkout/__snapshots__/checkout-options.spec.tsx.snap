// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CheckoutOptions Component matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="checkout-options"
  >
    <div
      class="grid grid--top-bottom grid-stretch-top"
    >
      <div
        class="grid__item"
      >
        <header>
          Select Redemption Amount
        </header>
      </div>
      <div
        class="grid__item"
      >
        <section
          class="options-list"
        >
          <section
            class="checkout-options-buttons"
          >
            <div
              class="grid"
            >
              <div
                class="grid__item"
              >
                <div
                  class="checkout-radio"
                >
                  <input
                    checked=""
                    id="checkout-option-button-775af84c-d388-4da7-b4cf-89f7fee1ebda"
                    name="giftcard-option"
                    type="radio"
                    value="1470"
                  />
                  <label
                    for="checkout-option-button-775af84c-d388-4da7-b4cf-89f7fee1ebda"
                  >
                    $14.70
                  </label>
                </div>
              </div>
              <div
                class="grid__item"
              >
                <div
                  class="checkout-radio"
                >
                  <input
                    id="checkout-option-button-21a2b9c5-f668-4c04-907b-019c53d72074"
                    name="giftcard-option"
                    type="radio"
                    value="2450"
                  />
                  <label
                    for="checkout-option-button-21a2b9c5-f668-4c04-907b-019c53d72074"
                  >
                    $24.50
                  </label>
                </div>
              </div>
              <div
                class="grid__item"
              >
                <div
                  class="checkout-radio"
                >
                  <input
                    id="checkout-option-button-24ed1a3d-8d86-4c7f-8bc8-1f4c3e6399b8"
                    name="giftcard-option"
                    type="radio"
                    value="4900"
                  />
                  <label
                    for="checkout-option-button-24ed1a3d-8d86-4c7f-8bc8-1f4c3e6399b8"
                  >
                    $49.00
                  </label>
                </div>
              </div>
            </div>
          </section>
        </section>
      </div>
      <div
        class="grid__item"
      >
        <section
          class="checkout-cost-summary"
        >
          <div
            class="grid"
          >
            <label>
              Redemption Amount
            </label>
            <div
              class="value"
            >
              $14.70
            </div>
            <label
              class="bonus-line"
            >
              Prizeout Bonus(2.04%)
            </label>
            <div
              class="value bonus-line"
            >
              $0.30
            </div>
            <label>
              You Get
            </label>
            <div
              class="value"
            >
              15.00
            </div>
          </div>
        </section>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
